stan/math/prim/scal/meta/and.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef STAN_MATH_PRIM_SCAL_META_AND_HPP
+#define STAN_MATH_PRIM_SCAL_META_AND_HPP
+
+#include <type_traits>
+
+namespace stan {
+namespace math {
+/**
+ * A helper metaprogram to peform conjuction
```

Comment:
In general, you don't need to say things like "a helper metaprogram to".  Just say

```
Extends std::true_type when instantiated with zero or more template parameters,
all of which extend the std::true_type.
```

At least I think that's what it's doing.  And I don't think the definition of `and` should link to `or` for an example.